src/EditorFeatures/Core/Implementation/IntelliSense/QuickInfo/IntellisenseQuickInfoBuilder.cs

```diff
@@ -47,18 +55,90 @@ internal static IntellisenseQuickInfoItem BuildItem(ITrackingSpan trackingSpan,
                 quickInfoItem.Sections.Where(s => s.Kind != QuickInfoSectionKinds.Description)
                                       .Select(BuildClassifiedTextElement));
 
+            // build text for RelatedSpan
+            if (quickInfoItem.RelatedSpans.Any())
+            {
+                elements.Add(await BuildTextElementForRelatedSpans(quickInfoItem, snapshot, document, cancellationToken).ConfigureAwait(false));
+            }
+
             var content = new ContainerElement(
                                 ContainerElementStyle.Stacked,
                                 elements);
 
             return new IntellisenseQuickInfoItem(trackingSpan, content);
         }
 
+        private static async Task<ClassifiedTextElement> BuildTextElementForRelatedSpans(CodeAnalysisQuickInfoItem quickInfoItem, ITextSnapshot snapshot, Document document, CancellationToken cancellationToken)
+        {
+            var classificationService = document.Project.LanguageServices.GetService<IClassificationService>();
+
+            var classifiedSpans = new List<ClassifiedSpan>();
+            var semanticSpans = new List<ClassifiedSpan>();
+
+            foreach (var span in quickInfoItem.RelatedSpans)
+            {
+                await classificationService.AddSyntacticClassificationsAsync(document, span, classifiedSpans, cancellationToken).ConfigureAwait(false);
+                await classificationService.AddSemanticClassificationsAsync(document, span, semanticSpans, cancellationToken).ConfigureAwait(false);
```

Comment:
Does this mean the tooltip won't appear until semantic classification is completed? That seems bad, any delay on tooltip is noticable.

The current behavior is to show without classification first and then add it later:
![image](https://user-images.githubusercontent.com/11444821/41436605-7a626f78-7022-11e8-9613-13e158aca6b6.png)

Reply:
let's tag @olegtk editor team for feedback. 
but I don't think this is serious enough to block moving to new API. and if we get enough feedbacks, might be added to v2 API.

tagging @jinujoseph  as well.

Reply:
@olegtk for the font style, you can actually get font style from format map and use same font type in quick info right? just a bug or is it intentional?

Reply:
for colorizer, wondering what is actual perf/time? the new way might be actually faster since span it asking is really small.

Reply:
We had a very very long list of requirements with the new generation of quick info APIs. Namely, they had to be VS Mac compatible, but still capable of hosting custom UX etc. I relied on Roslyn UX for most of my requirements.

Quick info API is pushing towards a data API. It's meant to aggregate and show messages/icons from a myriad of different sources, and as such, it's styled consistent with other IDE tooltips, using the standard tooltips font.

Previously, the font was monospace because we were hosting a text view. Text views can still be hosted in quick info via some hackery but like the EventHookup feature, this isn't really what quick info is intended to be used for.

FWIW, the IToolTipsService is a UI thread service that will pop up the same tooltip at a span with the same data object or even WPF. It's also a xplat safe type, so you can use in on Mac, so long as you don't take any direct WPF dependencies in that layer.